Good Coding Practices/src/js/scripts.js

```diff
@@ -1,44 +1,123 @@
+    this.el.style.top = `${randY}px`;
+  };
+  const setColor = function (color) {
+    this.el.style.background = color;
+    this.el.style.border = "1px solid " + color;
   };
   return {
     init,
-    moveDelta
+    moveDelta,
+    resetLocation,
+    randomLocation,
+    setColor
   };
 })();
 
+const Api = (function () {
+  let data = null;
+  const retriveData = function (url) {
+    fetch(url).then((res) => {
+      return res.json();
+    }).then((myJson) => {
+      this.data = JSON.stringify(myJson);
+      this.data = JSON.parse(this.data);
+      Api.setColorByCurrentTemp();
+    });
+  }
+  const setColorByCurrentTemp = function () {
+    const BLUE = "blue";
+    const GREEN = "green";
+    const YELLOW = "yellow";
+    const RED = "red";
+    if (this.data.current.temp_c <= 10) {
+      piece.setColor(BLUE);
+    }
+    else if (this.data.current.temp_c >= 11 && this.data.current.temp_c <= 20) {
+      piece.setColor(GREEN);
+    }
+    else if (this.data.current.temp_c >= 21 && this.data.current.temp_c <= 30) {
+      piece.setColor(YELLOW);
+    }
+    else {
+      piece.setColor(RED);
+    }
+  }
+  return {
+    retriveData,
+    setColorByCurrentTemp
+  }
+})();
+
 function handleClick(ev) {
   piece.moveDelta(parseInt(this.dataset.dx), parseInt(this.dataset.dy));
 }
 
 function init() {
+  const NUM_0 = 0;
+  const NUM_100 = 100;
+  const NUM_MINUS_100 = -100;
   const $btnUp = document.getElementById("btn-up");
-  $btnUp.dataset.dx = 0;
-  $btnUp.dataset.dy = -100;
-  $btnUp.addEventListener("click", handleClick);
+  setDxDy($btnUp, NUM_0, NUM_MINUS_100);
   const $btnRight = document.getElementById("btn-right");
-  $btnRight.dataset.dx = 100;
-  $btnRight.dataset.dy = 0;
-  $btnRight.addEventListener("click", handleClick);
+  setDxDy($btnRight, NUM_100, NUM_0);
   const $btnDown = document.getElementById("btn-down");
-  $btnDown.dataset.dx = 0;
-  $btnDown.dataset.dy = 100;
-  $btnDown.addEventListener("click", handleClick);
+  setDxDy($btnDown, NUM_0, NUM_100);
   const $btnLeft = document.getElementById("btn-left");
-  $btnLeft.dataset.dx = -100;
-  $btnLeft.dataset.dy = 0;
-  $btnLeft.addEventListener("click", handleClick);
+  setDxDy($btnLeft, NUM_MINUS_100, NUM_0);
+  addClickEventsMoveCircle($btnUp, $btnRight, $btnLeft, $btnDown);
+  const $btnReset = document.getElementById("btn-reset");
+  $btnReset.addEventListener("click", () => {
+    piece.resetLocation(NUM_0, NUM_0);
+  });
+  const $randomize = document.getElementById("btn-randomize");
+  $randomize.addEventListener("click", () => {
+    piece.randomLocation();
+  });
+}
+
+function setDxDy(el, dx, dy) {
+  el.dataset.dx = dx;
+  el.dataset.dy = dy;
+
+}
+function addClickEventsMoveCircle(...control) {
+  control.forEach((item) => {
+    item.addEventListener("click", handleClick);
+  });
 }
 
 window.addEventListener("DOMContentLoaded", event => {
+  const URL = "http://api.apixu.com/v1/current.json?key=dda6e762ae4f41efb7e173552192204&q=tel%20aviv";

```

Comment:
API URL should be part of the API knowledge, not the entire app.
Also, key should be a separate constant